kyaml/yaml/walk/associative_sequence.go

```diff
@@ -97,6 +103,42 @@ func validateKeys(valuesList [][]string, values []string, keys []string) ([]stri
 	return validKeys, validValues
 }
 
+func parseValues(valuesList [][]string) [][]string {
+	for i, values1 := range valuesList {
+		for j, values2 := range valuesList {
+			if matched, values := match(values1, values2); matched {
+				valuesList[i] = values
+				valuesList[j] = values
+			}
+		}
+	}
+	return valuesList
+}
+
+func match(values1 []string, values2 []string) (bool, []string) {
```

Comment:
same here, although this may not need as much explanation if the method above is explained